model/jpa/src/main/java/org/keycloak/models/jpa/entities/RoleEntity.java

```diff
@@ -99,6 +98,9 @@ public class RoleEntity {
     @JoinTable(name = "COMPOSITE_ROLE", joinColumns = @JoinColumn(name = "COMPOSITE"), inverseJoinColumns = @JoinColumn(name = "CHILD_ROLE"))
     private Set<RoleEntity> compositeRoles;
 
+    @ManyToMany(mappedBy = "compositeRoles", cascade = {})
```

Comment:
Should we set `fetch = FetchType.LAZY` to this relationship? We usually don't need to fetch this when using roles.

Reply:
At least when debugging it, it seems that it was lazy already that is fetching those items when accessing them. 

Added this in the next commit to enforce it. Thank you for spotting this.